### PR TITLE
Unify serialization

### PR DESCRIPTION
**Pull request by AlexanderSaydakov**
Simplified and unified serialization and deserialization except HLL, where it is quite different form everything else, and will take a separate effort.

### CODE REVIEW

**Comment by jmalkin** [comment]
I think we need to figure out what happened to code coverage.

File: kll/include/kll_sketch_impl.hpp [hunk 1]
```diff
@@ -388,28 +388,28 @@ template<typename T, typename C, typename S, typename A>
...
-    os.write((char*)&n_, sizeof(n_));
-    os.write((char*)&min_k_, sizeof(min_k_));
-    os.write((char*)&num_levels_, sizeof(num_levels_));
-    os.write((char*)&unused, sizeof(unused));
-    os.write((char*)levels_.data(), sizeof(levels_[0]) * num_levels_);
+    write(os, n_);
+    write(os, min_k_);
+    write(os, num_levels_);
+    write(os, unused);
+    os.write(reinterpret_cast<const char*>(levels_.data()), sizeof(levels_[0]) * num_levels_);
```
Comment (jmalkin) [inline kll/include/kll_sketch_impl.hpp:412]: Feels weird to have an os.write in the middle of this

Comment (AlexanderSaydakov) [reply kll/include/kll_sketch_impl.hpp:412]: this is a different case. maybe we can have another template for this as well

**Review by jmalkin — APPROVED** [review]
I like how it collapses the
```
declare variable
is.read(reinterpret_cast<>())
```
code blocks into something really easy to read.

I think it's eventually worth adding a template for the remaining cases, but this is still a nice improvement.

**Pull request merged**